cube/CMakeLists.txt

```diff
@@ -300,14 +301,15 @@ target_include_directories(vkcubepp PRIVATE .)
 target_compile_definitions(vkcubepp PRIVATE VK_NO_PROTOTYPES)
 
 if(APPLE)
-    # Keep RPATH so fixup_bundle can use it to find libraries
-    set_target_properties(vkcubepp PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
-    install(TARGETS vkcubepp BUNDLE DESTINATION "cube")
-    # Fix up the library references to be self-contained within the bundle.
-    install(CODE "
-        include(BundleUtilities)
-        fixup_bundle(\${CMAKE_INSTALL_PREFIX}/cube/vkcubepp.app \"\" \"\")
-        ")
+    install(
+        TARGETS vkcubepp
+        # Install runtime dependencies like the Vulkan::Loader so the app is self-contained
+        RUNTIME_DEPENDENCIES DESTINATION "cube/vkcubepp.app/Contents/Frameworks/"
+        BUNDLE DESTINATION "cube"
+    )
```

Comment:
Note until KhronosGroup/Vulkan-Hpp/pull/1755 is merged this will fail for vkcubepp since it uses the hpp dynamic loader.

Only `libvulkan.1.dylib` and `libvulkan.1.3.274.dylib` are automatically being detected as a runtime dependency by cmake. And currently hpp only checks for `libvulkan.dylib`

While Volk checks for libvulkan.dylib and libvulkan.1.dylib

Which is why vkcube installation will now work but vkcubepp doesn't. When I manually added detection for libvulkan.1.dylib vkcubepp also worked